src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+// import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal';
+import { SelectedStatus } from './types/SelectedStatus';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | 0>(0);
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedStatus, setSelectedStatus] = useState(SelectedStatus.ALL);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [hasErrorFromServer, setHasErrorFromServer] = useState(false);
+
+  const selectedTodo = useMemo((): Todo | undefined => {
+    return todos.find(todo => todo.id === selectedTodoId);
+  }, [todos, selectedTodoId]);
+
+  const fetchTodos = async () => {
+    setHasErrorFromServer(false);
+    try {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+      setVisibleTodos(todosFromServer);
+      setIsLoading(false);
+    } catch {
+      setHasErrorFromServer(true);
+      setIsLoading(false);
+    }
+  };
+
+  fetchTodos();
```

Comment:
check the place, where you need to call `fetchTodos` function
looks like with this code it can cause infinite call cycle, because this function will be called for each component rerendering